tools/picard.py

```diff
@@ -536,4 +536,35 @@ def execute(
             'RUN_BARCODE=' + run_barcode, 'LIBRARY_PARAMS=' + library_params
         ]
         PicardTools.execute(self, self.subtoolName, opts, JVMmemory)
+
+    def execute_single_sample(self, 
+        basecalls_dir,
+        output_file,
+        run_barcode, # this is a run-specific ID, ex. the flowcell ID
```

Comment:
Maybe rename this to `read_group_id_precursor` because that's basically what it's being used for in this use case? I mean ... it might even be safe to just always use the flowcell ID, I don't really see this being something the user would want to fiddle with.

Reply:
Sounds good.